stdlib/test/builtin/test_reversed.mojo

```diff
@@ -24,5 +24,62 @@ def test_reversed_list():
         check -= 1
 
 
+def test_reversed_dict():
+    var dict = Dict[String, Int]()
+    dict["a"] = 1
+    dict["b"] = 2
+    dict["c"] = 3
+    dict["d"] = 4
+    dict["a"] = 5
+
+    var keys = String("")
+    for key in reversed(dict):
+        keys += key[]
+
+    assert_equal(keys, "dcba")
+
+    # Order preserved
+
+    _ = dict.pop("a")
+    _ = dict.pop("c")
+
+    keys = String("")
+    for key in dict:
+        keys += key[]
+
+    assert_equal(keys, "bd")
+
+    keys = String("")
+    for key in reversed(dict):
+        keys += key[]
+
+    assert_equal(keys, "db")
+
+    # Empty dict is iterable
```

Comment:
**Suggestion** This is good as you have it, but it seems cleaner to not use the same local variable for testing this part.  Instead, just use a new local variable that is an empty dict rather than popping the elements.